### DISCUSSION TITLE
초기에는 template이 그려지지 않아 돔 요소가 없고, 이후 돔 요소가 생겼을 때 querySelector로 요소를 찾는 경우 크루 여러분들은 어떻게 하시나요?

### DISCUSSION THREAD

**Discussion by kkojae91** [woowacourse-study/2022-code-review-study#28] [Lotto] [answered]
저는 constructor에서 해당 돔 요소를 사용한다고 명시적으로 알리기 위해 null; 값을 주었는데, 

리뷰어님 : 저는 null값을 주는 코드를 제거해도 될거라고 생각했습니다. constructor 메서드는 클래스의 인스턴스 객체를 생성하고 초기화하는 메서드인데 이미 null값을 주는 변수들의 값은 undefined이기 때문입니다. ㅎ

이렇게 답변을 해주셨습니다. 크루분들의 의견이 궁금하네요!!

```js
constructor(app) {
  this.$app = getElement(app);
  render(this.$app, template.paymentSection);
  render(this.$app, createTemplate.paymentSection());

  this.$paymentInput = getElement(SELECTOR.PAYMENT_INPUT);
  this.$paymentButton = getElement(SELECTOR.PAYMENT_BUTTON);
  this.$coverTheBackground = null;
  this.$lottoResultSection = null;
  this.$restartButton = null;
  this.$lottoListToggleButton = null;
}
```

**Comment by lokba** [comment]
좋은 질문인 것 같아요.

그러면 저는 역질문을 드리고 싶습니다.

constructor()은 인스턴스를 초기화할때 호출되는 함수입니다.
돔요소가 렌더되고, 불러일으킬 element 요소를 왜 굳이 constructor()에서 초기화를 하는건가요?

돔요소가 렌더된 이후에, 멤버변수로 초기화하는 거에대해서는 어떻게 생각하시나요?🤔

**Comment by wonsss** [comment, marked as answer]
안녕하세요. 꼬재~
```  
this.$coverTheBackground = null;
this.$lottoResultSection = null;
this.$restartButton = null;
this.$lottoListToggleButton = null;
```
```  
this.$coverTheBackground;
this.$lottoResultSection;
this.$restartButton ;
this.$lottoListToggleButton;
```

리뷰어님께서 권하신 방법은, 위 코드처럼 constructor 안에서 null을 할당하고 있는 멤버변수들을 아래 코드처럼 null을 할당하지 말고 undefined로 두는 게 좋겠다는 말씀이신가요?

근데 null이나 undefined나 값이 없는 것은 같으니까, 굳이 null을 할당하지 않아도 괜찮은 것 아닌가 싶습니다.

**Comment by byhhh2** [comment]
저는 `constructor에서 해당 돔 요소를 사용한다고 명시적으로 알리기 위해` 요부분이 필요한가? 라는 의문이 들어요!
만약 로또를 구입하는 form이 존재하는 `PurchaseView`라면 form관련 요소들을 사용할 것이 전제되어 있지 않나요?

저는 특정 요소가 내부 메서드에서 여러번 사용될 때 constructor에서 선언해주었거든요!
